eventmesh/ApiServerEventMesh.js

```diff
@@ -0,0 +1,411 @@
+   * @param {string} opts.operationId
+   * @param {Object} opts.stateValue
+   */
+  updateOperationState(opts) {
+    logger.info('Updating Operation State with :', opts);
+    assert.ok(opts.operationName, `Property 'operationName' is required to update operation state`);
+    assert.ok(opts.operationType, `Property 'operationType' is required to update operation state`);
+    assert.ok(opts.operationId, `Property 'operationId' is required to update operation state`);
+    assert.ok(opts.stateValue, `Property 'stateValue' is required to update operation state`);
+    const patchedResource = {
+      'status': {
+        'state': opts.stateValue
+      }
+    };
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`${opts.operationName}.${CONST.APISERVER.HOSTNAME}`][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[opts.operationType](opts.operationId)
+        .status.patch({
+          body: patchedResource
+        }))
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  /**
+   * @param {string} opts.resourceId
+   * @param {string} opts.operationName
+   * @param {string} opts.operationType
+   * @param {Object} opts.value
+   */
+  updateLastOperation(opts) {
+    const patchedResource = {};
+    patchedResource.metadata = {};
+    patchedResource.metadata.labels = {};
+    patchedResource.metadata.labels[`last_${opts.operationName}_${opts.operationType}`] = opts.value;
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`${CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT}.${CONST.APISERVER.HOSTNAME}`][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[CONST.APISERVER.RESOURCE_NAMES.DIRECTOR](opts.resourceId)
+        .patch({
+          body: patchedResource
+        }))
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  /**
+   * @param {string} opts.resourceId
+   * @param {string} opts.operationName
+   * @param {string} opts.operationType
+   */
+  getLastOperation(opts) {
+    return Promise.try(() => apiserver.loadSpec())
+      .then(() => apiserver
+        .apis[`${CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT}.${CONST.APISERVER.HOSTNAME}`][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[CONST.APISERVER.RESOURCE_NAMES.DIRECTOR](opts.resourceId)
+        .get())
+      .then(json => json.body.metadata.labels[`last_${opts.operationName}_${opts.operationType}`])
+      .catch(err => {
+        return buildErrors(err);
+      });
+  }
+
+  /**
+   * @param {string} opts.operationName
+   * @param {string} opts.operationType
+   * @param {string} opts.operationId
+   * @param {Object} opts.value
+   */
+  patchOperationOptions(opts) {
+    return this.getOperationOptions(opts)
+      .then(res => {
+        let resJson = JSON.parse(res);
+        logger.info(`Patching ${resJson} with ${opts.value}`);
```

Comment:
Better use `res` here instead of `resJson` in the logger

Reply:
Done